src/mlx/warnings.py

```diff
@@ -12,46 +13,167 @@
+        self.reset()
 
-    def check_sphinx_warnings(self, line):
+    def reset(self):
+        self.count = 0
+        self.warn_min = 0
+        self.warn_max = 0
+
+    def check(self, line):
         '''
         Function for counting the number of sphinx warnings in a logfile.
         The function returns the number of warnings found
         '''
-        if re.search(sphinx_pattern, line):
-            self.sphinx_counter += 1
+        self.count += len(re.findall(self.pattern, line))
+
+    def set_maximum(self, maximum):
+        if self.warn_min == 0:
+            self.warn_max = maximum
+            return 0
+        elif self.warn_min > maximum:
+            print("Invalid argument: mininum limit ({min}) is higher than maximum limit ({max}). Cannot enter {value}". format(min=self.warn_min, max=self.warn_max, value=maximum))
+            return errno.EINVAL
+        else:
+            self.warn_max = maximum
+            return 0
+
+    def get_maximum(self):
+        return self.warn_max
+
+    def set_minimum(self, minimum):
+        if minimum > self.warn_max:
+            print("Invalid argument: mininum limit ({min}) is higher than maximum limit ({max}). Cannot enter {value}". format(min=self.warn_min, max=self.warn_max, value=minimum))
+            return errno.EINVAL
+        else:
+            self.warn_min = minimum
+            return 0
+
+    def get_minimum(self):
+        return self.warn_min
+
+    def return_count(self):
+        print("{count} {name} warnings found".format(count=self.count, name=self.name))
+        return self.count
+
+    def return_check_limits(self):
+        if self.count > self.warn_max:
+            print("Number of warnings ({count}) is higher than the maximum limit ({max}). Returning error code 1.".format(count=self.count, max=self.warn_max))
+            return 1
+        elif self.count < self.warn_min:
+            print("Number of warnings ({count}) is lower than the minimum limit ({min}). Returning error code 1.".format(count=self.count, min=self.warn_min))
+            return 1
+        else:
+            print("Number of warnings ({count}) is between limits {min} and {max}. Well done.".format(count=self.count, min=self.warn_min, max=self.warn_max))
+            return 0
+
+
+class SphinxChecker(WarningsChecker):
+
+    def __init__(self):
+        super(SphinxChecker, self).__init__(name='sphinx', pattern=sphinx_pattern)
 
-    def return_sphinx_warnings(self):
-        print("{count} sphinx warnings found".format(count=self.sphinx_counter))
-        return self.sphinx_counter
 
-    def check_doxygen_warnings(self, line):
+class DoxyChecker(WarningsChecker):
+
+    def __init__(self):
+        super(DoxyChecker, self).__init__(name='doxygen', pattern=doxy_pattern)
+
+
+class JUnitChecker(WarningsChecker):
+
+    def __init__(self):
+        super(JUnitChecker, self).__init__(name='junit', pattern=junit_pattern)
+
+
+class WarningsPlugin:
+    Sphinx = SphinxChecker()
```

Comment:
The higher layer is the ```WarningsChecker``` class. This one has the min and max values, so they can be adapted using the checkerlist variable. 

If you still need to be able to access the subclasses in this way, this is a sign that you're doing polymorphism wrong, and something is wrong in the architecture. We shouldn't have to touch these checkers after they have been instantiated (any configuration should be done when the check is instantiated IMO)